R/fileupload.R

```diff
@@ -94,7 +94,7 @@ FileUploadContext <- R6Class(
     },
     createUploadOperation = function(fileInfos) {
       while (TRUE) {
-        id <- paste(as.raw(p_runif(12, min=0, max=0xFF)), collapse='')
+        id <- createUniqueId(12)
         private$ids <- c(private$ids, id)
```

Comment:
This is a separate issue, but it seems to me that the `id` shouldn't be recorded here. Instead, it should be recorded after the `dir.create()` succeeds. Although realistically, there's no need to attempt the `dir.create` and then loop if it fails, because the probability of a name collision is so low. It seems more likely that the `dir.create` would fail due to a permission problem, and then this code would get stuck in an infinite loop.